### PR TITLE
Temporary mutability idiom

### PR DESCRIPTION
**Pull request by kriomant**

### CODE REVIEW

File: idioms/temporary-mutability.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## Advantages
+
+Compiler ensures that you don't accidentally mutate data after some point.
+
+
+## Disadvantages
+
+Nested block requires additional indentation of block body.
+One more line to return data from block or rebind variable (but it worth it).
```
[the remark this answers is not in the document]
Comment (kriomant) [reply idioms/temporary-mutability.md]: Done.

File: idioms/temporary-mutability.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Temporary mutability
+
+## Description
+
+Often it is necessary to prepare and process some data, but after that data are only inspected
+and never modified. It would be cool to turn mutable variable into immutable one.
```
[the remark this answers is not in the document]
Comment (kriomant) [reply idioms/temporary-mutability.md]: Fixed.

File: idioms/temporary-mutability.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+It can be done either by processing data within nested block or by rebinding variable.
```
[the remark this answers is not in the document]
Comment (kriomant) [reply idioms/temporary-mutability.md]: Fixed.

**Comment by kriomant** [comment]
I've fixed issues, please look

**Comment by lfairy** [comment]
Thanks!

**Pull request merged**